### ISSUE TITLE
Datagrid column should support aliasing

### ISSUE THREAD

**Issue by prakhargupta1** [mui/mui-toolpad#1561]
### Duplicates

- [X] I have searched the existing issues

### Latest version

- [X] I have tested the latest version

### Summary 💡

Problem: From a datagrid cell, I want to take the end-user to another page of the app. Example app and screenshot below, the column name is sales_rep.

<img width="816" alt="Screenshot 2023-01-16 at 1 50 32 PM" src="https://user-images.githubusercontent.com/92228082/212631120-e44e6106-81a0-4a8d-98c1-91fbd5eb2b6c.png">
 
App link: https://master--toolpad.mui.com/_toolpad/app/clbxozkn30001vg9jj6x4k742/pages/59i3m1g

Now, instead of showing the link, I want to show `Open` for all rows. 
Even better would be, if I can make the adjacent column `sales_rep_employee_number` a hyperlink that takes a custom link based on the employee id.

Solution: Columns should support creating alias.

Benchmarks: Retool, appsmith to be added

**Comment by apedroferreira** [comment]
> If we have a "link" column type, I think it's not too far fetched to have it support both url and text.

Yeah, in my opinion sounds like a common-enough use case that we should support it.
